src/nylas-connection.ts

```diff
@@ -246,43 +246,30 @@ export default class NylasConnection {
           }
 
           if (response.status > 299) {
-            return response
-              .clone()
-              .json()
-              .then(body => {
+            return response.text().then(body => {
+              try {
+                const jsonError = JSON.parse(body);
```

Comment:
So `jsonError` here would just be the parsed body, why do we expect it to be an error?
`JSON.parse()` throws an exception if there is an error with the provided JSON, so the catch would be handling that case already.

Reply:
Oh I misunderstood the variable naming, this is just the parsed error object because the response error code was `> 299`. Maybe consider renamind this variable.